cleverhans/attacks.py

```diff
@@ -592,20 +605,23 @@ def body(i, adv_x):
 
     _, adv_x = tf.while_loop(cond, body, [tf.zeros([]), adv_x], back_prop=True)
 
-    asserts = []
 
     # Asserts run only on CPU.
```

Comment:
It's not necessary to merge this PR, but if you feel like it you could change these lines to just use `utils_tf.assert*` rather than the explicit `with tf.device`

Reply:
I have to fix the crash anyway, so I'll just add it to this PR.